lib/renderer/api/desktop-capturer.js

```diff
@@ -1,46 +1,43 @@
-const ipcRenderer = require('electron').ipcRenderer
-const nativeImage = require('electron').nativeImage
+const {ipcRenderer, nativeImage} = require('electron')
 
-var nextId = 0
-var includes = [].includes
+const includes = [].includes
+let currentId = 0
 
-var getNextId = function () {
-  return ++nextId
+const incrementId = () => {
+  currentId += 1
+  return currentId
 }
 
 // |options.type| can not be empty and has to include 'window' or 'screen'.
-var isValid = function (options) {
-  return ((options != null ? options.types : void 0) != null) && Array.isArray(options.types)
+function isValid (options) {
+  const types = (options != null) ? options.types : undefined
+  return (types != null) && Array.isArray(options.types)
 }
 
 exports.getSources = function (options, callback) {
-  var captureScreen, captureWindow, id
-  if (!isValid(options)) {
-    return callback(new Error('Invalid options'))
-  }
-  captureWindow = includes.call(options.types, 'window')
-  captureScreen = includes.call(options.types, 'screen')
+  if (!isValid(options)) return callback(new Error('Invalid options'))
```

Comment:
_(style)_ I find this practice of having if ... action on the same line very difficult to read; e.g. when scanning for the return points in a function I would probably miss this.

That's just a personal preference, though -- do we have a JS style guide that we follow in Electron?

Reply:
> do we have a JS style guide that we follow in Electron?

https://github.com/electron/electron/blob/master/docs/development/coding-style.md#javascript

No hard and fast rule about one-liners. In general I think we use them as long they fit in 80 characters-ish.

Reply:
Yep, looks like [Standard JS](https://github.com/standard/standard/blob/master/docs/RULES-en.md) is explicitly OK with it.

nevermind my personal preference then :neutral_face:

Reply:
@ckerr, your personal preference makes perfect sense )
Two expressions on the same line is generally a bad idea. When one of the expressions has to be changed the whole line is marked as "changed" and it means that a) the diff is less readable, and b) git blame becomes screwed up.

I personally always use braces and for instance Google JavaScript Style Guide explicitly [requires them](https://google.github.io/styleguide/jsguide.html#formatting-braces) for all control structures.